File: emp_frontend/src/app/models/leave.ts
import {Employee} from "./employee";

export class Leave {
  id: number;
  description: string;
  employee!: Employee; // Reference to Employee model
  startDate: string; // ISO 8601 format
  endDate: string; // ISO 8601 format
  status: string;


  constructor() {
    this.id = 0;
    this.description = "";
    //this.employee = "";
    this.startDate = "";
    this.endDate = "";
    this.status = "";
  }
}

File: emp_backend/src/main/java/com/example/demo/model/Leave.java
package com.example.demo.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity(name = "leave")
public class Leave {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String description;
    @ManyToOne
    @JoinColumn(name = "employee_id", referencedColumnName = "id")
    private Employee employee;
    private LocalDate startDate;
    private LocalDate endDate;
    private String status;
}
